connect/api/src/main/java/org/apache/kafka/connect/data/Field.java

```diff
@@ -71,7 +71,10 @@ public boolean equals(Object o) {
 
     @Override
     public int hashCode() {
-        return Objects.hash(name, index, schema);
+        int result = index;
+        result = 31 * result + (name != null ? name.hashCode() : 0);
+        result = 31 * result + (schema != null ? schema.hashCode() : 0);
```

Comment:
Clearly this is the expensive part of the computation. I was looking through the code to find instances where we are using `Field` as the key in a map, but didn't see any. Ideally we would prefer to use the field name instead of the field in these situations, which is what is done in `ConnectSchema`. Perhaps you can clarify which use of `Field` is causing the performance issue? Or is it inside a connector?

Reply:
Field is used only as part of ConnectSchema.
So calculation of ConnectSchema.hashCode triggers Field.hashCode.